services/frontend/adapters/inbound/secret_watcher.go

```diff
@@ -0,0 +1,216 @@
+							// File exists, save it to Context-box
+							log.Debug().Msgf("Assuming file %s from secret %s was modified", fileName, secret.Name)
+							manifest, err := sw.getRawManifest(file, secret.Name, fileName)
+							if err != nil {
+								log.Err(err).Msgf("Failed to decode file %s from secret %s, skipping...", fileName, secret.Name)
+								continue
+							}
+							sw.usecases.SaveChannel <- manifest
+						} else {
+							// File does not exists, trigger deletion
+							log.Debug().Msgf("Assuming file %s from secret %s was removed", fileName, secret.Name)
+							manifest, err := sw.getRawManifest(file, secret.Name, fileName)
+							if err != nil {
+								log.Err(err).Msgf("Failed to decode file %s from secret %s, skipping...", fileName, secret.Name)
+								continue
+							}
+							sw.usecases.DeleteChannel <- manifest
+						}
+					}
+				}
+				// Check if any files from secret needs to be saved in DB
+				for name, file := range secret.Data {
+					// Manifest not in the database yet, save
+					if _, ok := inDB[name]; !ok {
+						manifest, err := sw.getRawManifest(file, secret.Name, name)
+						if err != nil {
+							log.Err(err).Msgf("Failed to decode file %s from secret %s, skipping...", name, secret.Name)
+							continue
+						}
+						sw.usecases.SaveChannel <- manifest
+					}
+				}
+			// Deleted secret
+			case watch.Deleted:
+				// All manifest in the secret were deleted.
+				for name, file := range secret.Data {
+					manifest, err := sw.getRawManifest(file, secret.Name, name)
+					if err != nil {
+						log.Err(err).Msgf("Failed to decode file %s from secret %s, skipping...", name, secret.Name)
+						continue
+					}
+					sw.usecases.DeleteChannel <- manifest
+				}
+			case watch.Bookmark, watch.Error:
+				// Due to golangci-lint, this case has to be included.
+				log.Debug().Msgf("Got event Bookmark or Error; not supported")
+			}
+		}
+	}
+	return rsv
+}
+
+// getRawManifest returns usecases.RawManifest from the given file.
+func (sw *SecretWatcher) getRawManifest(file []byte, secretName, fileName string) (*usecases.RawManifest, error) {
+	decoded, err := sw.decodeContent(file)
+	if err != nil {
+		return nil, err
+	}
+	return &usecases.RawManifest{
+		Manifest:   decoded,
+		SecretName: secretName,
+		FileName:   fileName,
+	}, nil
+}
+
+// decodeContent tries to decode base64 files and returns decoded version. If decoding fails,
+// it might assume the content is not base64 based and returns original content.
+func (sw *SecretWatcher) decodeContent(content []byte) ([]byte, error) {
+	decoded := make([]byte, len(content)*(4/3))
+	if _, err := base64.StdEncoding.Decode(decoded, content); err != nil {
+		// Cant use errors.Is() as base64 package builds error dynamically in base64.CorruptInputError()
+		if strings.Contains(err.Error(), "illegal base64 data") {
+			log.Debug().Msgf("File not base64 compatible, assuming it is string data")
+			return content, nil
+		}
```

Comment:
Why can't we use `errors.Is(err, base64.CorruptInputError)` ? 

The errors return from the function in the base64 package seem to always return `CorruptInputError`

Reply:
Oh, my bad, I have looked at `base64.CorruptInputError` and it's defined as `int64`. But yes, it implements `error` interface which I missed, that's why I used this.